framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/ConstraintSet.java

```diff
@@ -334,9 +373,17 @@ public BoundSet reduceOneStep(Java8InferenceContext context) {
       }
       this.addAll(((ReductionResultPair) result).constraintSet);
     } else if (result instanceof TypeConstraint) {
-      this.add((Constraint) result);
+      // Add the new constraints to the beginning of the list so they are reduced first. This is
+      // because each constraint is supposed to be fully resolved before moving onto another one.
```

Comment:
Is this ordering related to the greediness I asked about earlier?  That greedy algorithm might resolve later variables in the list first; is that OK?

Reply:
No, this code is reducing constraints not resolving variables.   (The use of `resolved` in this comment is incorrect and I've rewritten it.)